Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -26,21 +26,125 @@
 #include "CompressionStreamEncoder.h"
 
 #include "BufferSource.h"
+#include "Exception.h"
 #include <JavaScriptCore/GenericTypedArrayViewInlines.h>
 #include <JavaScriptCore/JSCInlines.h>
 #include <JavaScriptCore/JSGenericTypedArrayViewInlines.h>
 
 namespace WebCore {
 
-RefPtr<Uint8Array> CompressionStreamEncoder::encode(const BufferSource&& input)
+ExceptionOr<RefPtr<Uint8Array>> CompressionStreamEncoder::encode(const BufferSource&& input)
 {
-    UNUSED_PARAM(input);
-    return nullptr;
+    size_t length = input.length();
+    auto* data = input.data();
+
+    if (!data)
+        return Exception { TypeError, "No data provided"_s };
+
+    auto compressedDataCheck = compress(data, length);
+    if (compressedDataCheck.hasException())
+        return compressedDataCheck.releaseException();
+
+    auto compressedData = compressedDataCheck.returnValue();
+    if (!compressedData.size())
+        return nullptr;
+    
+    return Uint8Array::tryCreate(compressedData.data(), compressedData.size());
 }
 
-RefPtr<Uint8Array> CompressionStreamEncoder::flush()
+ExceptionOr<RefPtr<Uint8Array>> CompressionStreamEncoder::flush()
 {
-    return nullptr;
+    finish = true;
+
+    auto compressedDataCheck = compress(0, 0);
+    if (compressedDataCheck.hasException())
+        return compressedDataCheck.releaseException();
+    
+    auto compressedData = compressedDataCheck.returnValue();
+    if (!compressedData.size())
+        return nullptr;
+    
+    return Uint8Array::tryCreate(compressedData.data(), compressedData.size());
+}
+
+ExceptionOr<bool> CompressionStreamEncoder::initialize() 
+{
+    int result;
+
+    initailized = true;
+    zstream.opaque = 0;
+    zstream.zalloc = 0;
+    zstream.zfree = 0;
+
+    switch (m_format) {
+    // Values chosen here are based off
+    // https://developer.apple.com/documentation/compression/compression_algorithm/compression_zlib?language=objc
+    case Formats::CompressionFormat::Deflate:
+        result = deflateInit2(&zstream, 5, Z_DEFLATED, -15, 8, Z_DEFAULT_STRATEGY);
+        break;
+    case Formats::CompressionFormat::Zlib:
+        result = deflateInit2(&zstream, 5, Z_DEFLATED, 15, 8, Z_DEFAULT_STRATEGY);
+        break;
+    case Formats::CompressionFormat::Gzip:
+        result = deflateInit2(&zstream, 5, Z_DEFLATED, 15 + 16, 8, Z_DEFAULT_STRATEGY);
+        break;
+    default:
+        RELEASE_ASSERT_NOT_REACHED();
+        break;
+    }
+
+    if (result != Z_OK)
+        return Exception { TypeError, "Initialization Failed."_s };
+
+    return true;
 }
 
+ExceptionOr<Vector<uint8_t>> CompressionStreamEncoder::compress(const uint8_t* input, const size_t inputLength)
+{
+    size_t index = 0, totalSize = 0;
+    size_t allocateSize = (inputLength < 16384) ? 16384 : inputLength;
```

Comment:
The magic number 16384 should be declared as a constant somewhere to explain what it is.